lang/java/avro/src/main/java/org/apache/avro/Schema.java

```diff
@@ -345,6 +379,9 @@ public String getFullName() {
     return getName();
   }
 
+  public void terminateInit() {
```

Comment:
What does this method do?

I understand methods like `hasChild`, `visitHierarchy` and `findInHierarchy` above (though they also need javadoc), but I cannot understand the what&why of this one.

Reply:
It's a kind of PostConstruct, for SchemaRecord, once all schemas has been built and list of fields set.
But, indeed, i will see if we can do without it

Reply:
Ah, sort of like how the fields are marked immutable once set for the first time? (and any field use before they are set yields an exception)

How about this idea:
1. in the constructor, mark if the record schema will have children (if the parameter is omitted, it will not)
2. have a method where you can set all child schemata, once

This behaves like `setFields`: it allows us both to finalise initialisation and makes the schema unusable for (de)serialisation until the child schemata have been set (so it cannot be forgotten).

Reply:
So, i finally find an easy way to remove this weird method.

Reply:
I saw; lazy initialisation. Also a good idea.